src/tree_cmd.cpp

```diff
@@ -189,17 +328,20 @@ static void PlaceTreeGroups(uint num_groups)
 	do {
 		TileIndex center_tile = RandomTile();
 
+		std::vector<BlobPosition> grove = createRandomStarShapedPolygon(GROVE_RADIUS, GROVE_RESOLUTION);
```

Comment:
You're recreating a polygon which is the same every time. This could be outside the do-loop.

Reply:
the same every time? i don't think so, it's a different one per group and makes several calls to `Random()`